src/ape/contracts/base.py

```diff
@@ -875,33 +879,19 @@ def from_receipt(cls, receipt: ReceiptAPI, contract_type: ContractType) -> "Cont
             contract_type=contract_type,
             txn_hash=receipt.txn_hash,
         )
-        instance._cached_receipt = receipt
+
+        # Cache creation.
+        creation = ContractCreation.from_receipt(receipt)
+        cls.chain_manager.contracts._local_contract_creation[address] = creation
+
         return instance
 
     @property
-    def receipt(self) -> ReceiptAPI:
+    def creation(self) -> Optional[ContractCreation]:
```

Comment:
suggestion: I still think that this slightly more verbose property would be less likely to cause a conflict
```suggestion
    def creation_metadata(self) -> Optional[ContractCreation]:
```